tests/v2/validation/provisioning/rke2/delete_rke2_node_driver_cluster_test.go

```diff
@@ -0,0 +1,229 @@
+	user := &management.User{
+		Username: testuser,
+		Password: testpassword,
+		Name:     testuser,
+		Enabled:  &enabled,
+	}
+
+	newUser, err := users.CreateUserWithRole(client, user, "user")
+	require.NoError(r.T(), err)
+
+	newUser.Password = user.Password
+
+	standardUserClient, err := client.AsUser(newUser)
+	require.NoError(r.T(), err)
+
+	r.standardUserClient = standardUserClient
+}
+
+func (r *RKE2NodeDriverDeletingTestSuite) CreateAndDeleteRKE2Cluster(client *rancher.Client, provider Provider, nodesAndRoles []machinepools.NodeRoles, kubeVersion, cni string, externalNodeProvider *provisioning.ExternalNodeProvider) {
+	cloudCredential, err := provider.CloudCredFunc(client)
+	nodeNames := []string{}
+
+	clusterName := namegen.AppendRandomString(provider.Name.String())
+	generatedPoolName := fmt.Sprintf("nc-%s-pool1-", clusterName)
+	machinePoolConfig := provider.MachinePoolFunc(generatedPoolName, namespace)
+
+	machineConfigResp, err := client.Steve.SteveType(provider.MachineConfigPoolResourceSteveType).Create(machinePoolConfig)
+	require.NoError(r.T(), err)
+
+	machinePools := machinepools.RKEMachinePoolSetup(nodesAndRoles, machineConfigResp)
+
+	cluster := clusters.NewK3SRKE2ClusterConfig(clusterName, namespace, cni, cloudCredential.ID, kubeVersion, machinePools)
+
+	clusterResp, err := clusters.CreateK3SRKE2Cluster(client, cluster)
+	require.NoError(r.T(), err)
+
+	kubeProvisioningClient, err := r.client.GetKubeAPIProvisioningClient()
+	require.NoError(r.T(), err)
+
+	result, err := kubeProvisioningClient.Clusters(namespace).Watch(context.TODO(), metav1.ListOptions{
+		FieldSelector:  "metadata.name=" + clusterName,
+		TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+	})
+	require.NoError(r.T(), err)
+
+	checkFunc := clusters.IsProvisioningClusterReady
+
+	err = wait.WatchWait(result, checkFunc)
+	assert.NoError(r.T(), err)
+	assert.Equal(r.T(), clusterName, clusterResp.ObjectMeta.Name)
+	assert.Equal(r.T(), kubeVersion, cluster.Spec.KubernetesVersion)
+
+	clusterIDName, err := clusters.GetClusterIDByName(r.client, clusterName)
+	assert.NoError(r.T(), err)
+
+	fmt.Println("--------Checking if the nodes are ready-----------")
+	err = nodestat.IsNodeReady(client, clusterIDName)
+	require.NoError(r.T(), err)
+
+	machineResp, err := r.client.Steve.SteveType("cluster.x-k8s.io.machine").List(nil)
+	require.NoError(r.T(), err)
+	for _, machine := range machineResp.Data {
+		machineobj := &capi.Machine{}
+		err = v1.ConvertToK8sType(machine.Spec, &machineobj.Spec)
+		require.NoError(r.T(), err)
+		if machineobj.Spec.ClusterName == clusterName {
+			nodeNames = append(nodeNames, machineobj.Spec.InfrastructureRef.Name)
+		}
+	}
+
+	fmt.Println("---waiting for 1 minute before deleting the cluster---")
+	time.Sleep(1 * time.Minute)
+
+	fmt.Print("--------deleting cluster now-----------")
+	err = r.deleteCluster(clusterResp)
+	assert.NoError(r.T(), err)
+
+	fmt.Println("--------Checking if the nodes are deleted from AWS-----------")
```

Comment:
If this tests runs for all the different node providers, how would this test work if it was provisioned on azure, or linode? You're making a specific aws check.

Reply:
Agreed, this would only work for EC2.